includes/AcquiaLiftAPIV2.inc

```diff
@@ -470,7 +470,7 @@ class AcquiaLiftAPIV2 extends AcquiaLiftAPIBase implements AcquiaLiftLearnReport
     $message = $this->canonicalizeRequest($verb, $host, $path, $query_args, $authorization, $timestamp, $content_type, $payload_hash);
 
     // Always use binary output
-    $signature = base64_encode(hash_hmac('sha256', $this->api_private_key, (string) $message, TRUE));
+    $signature = base64_encode(hash_hmac('sha256', (string) $message, $this->api_private_key, TRUE));
```

Comment:
Here's an explanation as to why you need type-casting in PHP to be a string explicitly: http://stackoverflow.com/questions/13884803/why-is-hash-hmac-making-different-hashes-for-these-two-large-numbers

Same thinking applies, an integer will hash differently than an empty string and since FALSE in PHP is not a string and we can't know if the message will be empty or not, we need to convert it to a string explicitly. Not required for the private key as that always needs to be a string anyway and if that is empty, it will not be correct.